pulsar-broker/src/main/java/org/apache/pulsar/broker/service/nonpersistent/NonPersistentSubscription.java

```diff
@@ -518,4 +521,8 @@ public long getLastActive() {
     public void updateLastActive() {
         this.lastActive = System.currentTimeMillis();
     }
+
+    public Map<String, String> getSubscriptionProperties() {
+        return subscriptionProperties;
```

Comment:
better to return an immutable map here

Reply:
there is no need to wrap it. it is an internal method and it should be already immutable.
this method is supposed to be called in hotpaths and we have to reduce allocations as much as possible

Reply:
I have wrapped the properties with unmodifiabledMap